tests/cases/conformance/controlFlow/controlFlowStringIndex.ts

```diff
@@ -0,0 +1,6 @@
+// @strict: true
+type A = { [index: string]: number | null };
+declare const value: A;
+if (value.foo !== null) {
+    value.foo.toExponential()
```

Comment:
Could you add a `value.bar` inside the if statement, too? Just to verify that other properties aren't narrowed despite being the same index signature. (I don't think it'll be wrong, I just want to see it in the test)